contracts/utils/cryptography/ERC7913SignatureVerifierZKEmail.sol

```diff
@@ -0,0 +1,86 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.24;
+
+import {IDKIMRegistry} from "@zk-email/contracts/DKIMRegistry.sol";
+import {IVerifier} from "@zk-email/email-tx-builder/interfaces/IVerifier.sol";
+import {EmailAuthMsg} from "@zk-email/email-tx-builder/interfaces/IEmailTypes.sol";
+import {IERC7913SignatureVerifier} from "../../interfaces/IERC7913.sol";
+import {ZKEmailUtils} from "./ZKEmailUtils.sol";
+
+/**
+ * @dev ERC-7913 signature verifier that supports ZKEmail accounts.
+ *
+ * This verifier validates signatures produced through ZKEmail's zero-knowledge
+ * proofs which allows users to authenticate using their email addresses.
+ */
+abstract contract ERC7913SignatureVerifierZKEmail is IERC7913SignatureVerifier {
+    using ZKEmailUtils for EmailAuthMsg;
+
+    IVerifier private immutable _verifier;
+    uint256 private immutable _templateId;
+
+    constructor(IVerifier verifier_, uint256 templateId_) {
+        _verifier = verifier_;
+        _templateId = templateId_;
+    }
+
+    /// @dev An instance of the Verifier contract.
+    /// See https://docs.zk.email/architecture/zk-proofs#how-zk-email-uses-zero-knowledge-proofs[ZK Proofs].
+    function verifier() public view virtual returns (IVerifier) {
+        return _verifier;
+    }
+
+    /// @dev The command template of the sign hash command.
+    function templateId() public view virtual returns (uint256) {
+        return _templateId;
+    }
+
+    /**
+     * @dev Verifies a zero-knowledge proof of an email signature validated by a {DKIMRegistry} contract.
+     *
+     * The key format is ABI-encoded (IDKIMRegistry, bytes32) where:
+     * - IDKIMRegistry: The registry contract that validates DKIM public key hashes
```

Comment:
Is this a canonical address or one particular to the DKIM public key?

Reply:
Not really, they even suggest to deploy it yourself: https://docs.zk.email/email-tx-builder/quickstart#build-and-deploy

Reply:
> Is this a canonical address or one particular to the DKIM public key?

actually, we do envision this as a canonical address per chain, as this is the dkim registry that get's updated by our infrastructure. Users can override the dkim keys on the canonical registry if they want

Ofc users can still deploy their own but they would have to update it themselves.

This is not clear from our docs so we will update them!

Reply:
> actually, we do envision this as a canonical address per chain, as this is the dkim registry that get's updated by our infrastructure. 

This would be amazing! If that's the case we could update the `SignerZKEmail` and this `ERC7913SignatureVerifierZKEmail` to hardcode the address of the canonical registry in the `DKIMRegistry` function, similar to how we [hardcode the Entrypoint in Accounts](https://github.com/OpenZeppelin/openzeppelin-community-contracts/blob/master/contracts/account/Account.sol#L49). Ideally I would like developers to not have to worry about locating/deploying the registry

Reply:
yeah ok we could look at doing that. We can run a new deployment and provide an address that would be consistent across chains

Reply:
Opened an issue to track this: https://github.com/OpenZeppelin/openzeppelin-community-contracts/issues/140

Reply:
Sweet. We'll aim to do the multi-chain deploy by end of next week